ui/src/main/js/page/settings/standalone/SettingsEncryptionConfiguration.js

```diff
@@ -0,0 +1,93 @@
+    const [errors, setErrors] = useState(HttpErrorUtilities.createEmptyErrorObject());
+
+    const [passwordFromApiExists, setPasswordFromApiExists] = useState(false);
+
+    const fetchData = async () => {
+        const response = await ConfigurationRequestBuilder.createReadRequest(encryptionRequestUrl, csrfToken);
+        const data = await response.json();
+
+        if (data) {
+            setSettingsEncryptionConfig(data);
+            setPasswordFromApiExists(true);
+        } else {
+            setSettingsEncryptionConfig({});
+            setPasswordFromApiExists(false);
+        }
+    };
+
+    return (
+        <ConfigurationForm
+            csrfToken={csrfToken}
+            setErrors={(formErrors) => setErrors(formErrors)}
+            buttonIdPrefix={SETTINGS_INFO.key}
+            getRequest={fetchData}
+            deleteRequest={() => null}
+            createRequest={() => ConfigurationRequestBuilder.createUpdateWithoutIdRequest(encryptionRequestUrl, csrfToken, settingsEncryptionConfig, settingsEncryptionConfig)}
+            updateRequest={() => ConfigurationRequestBuilder.createUpdateWithoutIdRequest(encryptionRequestUrl, csrfToken, settingsEncryptionConfig, settingsEncryptionConfig)}
+            validateRequest={() => ConfigurationRequestBuilder.createValidateRequest(encryptionRequestUrl, csrfToken, settingsEncryptionConfig)}
+            readonly={readonly}
+            displaySave={displaySave}
+            displayTest={false}
+            displayDelete={false}
+            errorHandler={errorHandler}
+        >
+            <h2 key="settings-header">Encryption Configuration</h2>
+            <PasswordInput
+                id={SETTINGS_FIELD_KEYS.encryptionPassword}
+                name="encryptionPassword"
+                label="Encryption Password"
+                description="The password used when encrypting sensitive fields. Must be at least 8 characters long."
+                required
+                readOnly={readonly}
+                onChange={fieldModelUtilities.handleTestChange(settingsEncryptionConfig, setSettingsEncryptionConfig)}
+                value={settingsEncryptionConfig.encryptionPassword || undefined}
+                isSet={passwordFromApiExists}
+                errorName="encryptionPassword"
+                errorValue={errors.fieldErrors.encryptionPassword}
+            />
+            <PasswordInput
+                id={SETTINGS_FIELD_KEYS.encryptionGlobalSalt}
+                name="encryptionGlobalSalt"
+                label="Encryption Global Salt"
+                description="The salt used when encrypting sensitive fields. Must be at least 8 characters long."
+                required
+                readOnly={readonly}
+                onChange={fieldModelUtilities.handleTestChange(settingsEncryptionConfig, setSettingsEncryptionConfig)}
+                value={settingsEncryptionConfig.encryptionGlobalSalt || undefined}
+                isSet={passwordFromApiExists}
```

Comment:
You'd want to call your second password state boolean here which is based off the content returned from the fetch call